### ISSUE TITLE
Retroarch core has severe lag and slowdown 

### ISSUE THREAD

**Issue by pokemonfan6498** [bsnes-emu/bsnes#317]
i was testing the bsnes core and i noticed that it has severe lag which is strange because snes9x runs fine and that's not just regular snes roms and bs roms Gameboy roms running on sgb an sgb2 also have severe lag and cpu overclock (if on switch) doesn't help much

**Comment by pokemonfan6498** [comment]
really

**Comment by Kawa-oneechan** [comment]
Yeah, really.

**Comment by carmiker** [comment]
The way bsnes polls for input can lead to inherent lag depending on how/when the frontend updates the input state which bsnes is polling. There are options in RetroArch that work around this problem somewhat. If you are able to run at full speed (60fps), this is not a CPU usage problem. In RetroArch you can try switching the `Polling Behavior` option in the `Latency` menu to the opposite of whatever it is now, and see if that helps.

**Comment by hizzlekizzle** [comment]
yeah, "lag" can be ambiguous. If OP means "latency," then yeah, @carmiker 's suggestion could be worth checking out. If they mean "frame drops" it's definitely related to hardware requirements vs CPU grunt.

On my Nvidia Shield, I could run bsnes on regular SNES games at full speed, but I could never get higher than 45 fps with 2x hd-mode-7 scale. I don't recall how well it handled SGB. Regardless, I believe the Shield's clock speed is something like double the Switch's, so performance is likely to be worse on the Switch.

**Comment by pokemonfan6498** [comment]
but im using retroarch core how im supposed to edit that?

**Comment by alice-mkh** [comment]
I was suggesting to run the entire retroarch like that. But yeah, on switch you probably can't :/

**Comment by alice-mkh** [comment]
Yeah

What sucks is that it's seemingly impossible to set this variable programmatically - otherwise that would be an easy fix

The only thing that worked for me was setting it _before even loading the core_, from the frontend
